docker/builder/builder.go

```diff
@@ -52,9 +54,30 @@ func (d *DaemonBuilder) Build(ctx context.Context, imageName string) error {
 		return err
 	}
 	defer buildCtx.Close()
+	var l logger.Logger
+	if d.LoggerFactory != nil {
+		l = d.LoggerFactory.CreateBuildLogger(imageName)
+	}
 
 	var progBuff io.Writer = os.Stdout
```

Comment:
I haven't checked out this PR too much yet, but enforcing a logger factory sounds like a good idea to me